### PR TITLE
Add Configurations documents

### LINKED ISSUE #954: Add a configurations md file under docs
### Problem

Currently there's no one place that lists all the configuration env variable keys, their default values and descriptions.
This makes it difficult for devs to know what is set and what should be tweaked

### Solution

Add a doc under `/docs/configuration.md` to highlight the missing details and bring clarity
Doc should have a table that notes

- config name
- default value
- description

Note: inspiration drawn from efforts on https://github.com/hashgraph/hedera-mirror-node/blob/main/docs/configuration.md

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by Nana-EC**
**Description**:
- Add a configuration document at `docs/configuration.md`
  - highlight individual package values
  - highlight test configs
- Updated main README with link to configuration document

**Related issue(s)**:

Fixes #954 

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: docs/configuration.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+Unless you need to set a non-default value, it is recommended to only populate overridden properties in the custom `.env`.
+
+| Name                          | Default       | Description                                                   |
+|-------------------------------|---------------|---------------------------------------------------------------|
+| `CHAIN_ID`                    | ""            | The network chain id. Local and previewnet envs should use `0x12a` (298). Previewnet, Testnet and Mainnet should use `0x129` (297), `0x128` (296) and `0x127` (295) respectively. |
+| `HBAR_RATE_LIMIT_DURATION`    | "60000"       | hbar budget limit duration. This creates a timestamp, which resets all limits, when it's reached. Default is to 60000 (1 minute).   |
+| `HBAR_RATE_LIMIT_TINYBAR`    | "5000_000_000" | total hbar budget in tinybars.   |
+| `HEDERA_NETWORK`              | ""                         | Which network to connect to. Automatically populates the main node & mirror node endpoints. Can be `MAINNET`, `PREVIEWNET`, `TESTNET` or `OTHER`   |
```
Comment (lukelee-sl) [inline docs/configuration.md:32]: We give this example below.  `HEDERA_NETWORK={"127.0.0.1:50211":"0.0.3"}`.
Can we highlight what this means when the `HEDERA_NETWORK` is `OTHER`

File: docs/configuration.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+| `LOCAL_NODE`      | ""        | Flag if relay is hosted in the Hedera local node setup.   |
+| `E2E_RELAY_HOST`  | ""        | Remote relay url to point to.   |
+| `DEV_MODE`        | "false"   | Flag if relay should operate in developer optimization mode.   |
+
+For test context additional fields need to be set. The following example showcases a `hedera-local-node` instance (where values match those noted on [Local Node Network Variables](https://github.com/hashgraph/hedera-local-node#network-variables)
+
+```.env
+HEDERA_NETWORK={"127.0.0.1:50211":"0.0.3"}
+OPERATOR_ID_MAIN=0.0.2
+OPERATOR_KEY_MAIN=302e020100300506032b65700422042091132178e72057a1d7528025956fe39b0b847f200ab59b2fdd367017f3087137
```
Comment (lukelee-sl) [inline docs/configuration.md:82]: These are the values for local node.  Would it be helpful to highlight that for previewnet/testnet you can get these values from the [portal](http://portal.hedera.com/login)?

Comment (Nana-EC) [reply docs/configuration.md:82]: Will add

Comment (Nana-EC) [reply docs/configuration.md:82]: Done

File: docs/configuration.md (hunk 1 above)
Comment (Nana-EC) [reply docs/configuration.md]: Done

File: docs/configuration.md [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+| `MIRROR_NODE_LIMIT_PARAM`         | "100"         | The mirror node custom limit value to be set on GET requests. This optimizes the flow to reduce the number of calls made to the mirror node by setting a limit larger than it's default limit.   |
+| `MIRROR_NODE_RETRIES`             | "3"           | The maximum number of retries on a GET request to the mirror node when an acceptable error code is returned.   |
+| `MIRROR_NODE_RETRY_DELAY`         | "250"         | The dealy in ms between retry requests.   |
+| `MIRROR_NODE_URL`                 | ""            | The Mirror Node API endpoint. Official endpoints are Previewnet (https://previewnet.mirrornode.hedera.com/v1/api), Testnet (https://testnet.mirrornode.hedera.com/v1/api), Mainnet (https://mainnet-public.mirrornode.hedera.com/v1/api). See [Mirror Node REST API](https://docs.hedera.com/hedera/sdks-and-apis/rest-api)   |
+| `SDK_REQUEST_TIMEOUT`             | "10000"       | The complete timeout for running the SDK `execute()` method. This controls the GRPC channel timeout cofig when querying with network nodes.   |
```
Comment (ebadiere) [inline docs/configuration.md:60]: Spelling mistake, "cofig" when it should be "config".

Comment (Nana-EC) [reply docs/configuration.md]: Fixed

**Review by lukelee-sl — APPROVED** [review]
LGTM

**Pull request merged**